### DISCUSSION TITLE
Have two characters chat issue

### DISCUSSION THREAD

**Discussion by Moperius** [undreamai/LLMUnity#263] [Q&A] [answered]
Hi there. Awesome tool. Making a project with it and will try to give back some day.

Business time:
I have 2 LLMCharaters based on prefabs and an LLM, also based on a prefab.
I instantiate them, tie them to the LLM, set name and prompt, and start a conversation by using "_ = Chat..." statements in each other's completionCallback methods with a condition to avoid infinity.

This used to work with an older version of the library, but I updated a few days ago and I get stuck after the first Character finishes their line. The completionCallback get executed, but the chat of the second character never starts generating. I then stop the project and see in the console that the second character's completionCallback gets called as the project closes

How did you get two agents to talk to each other? Do you see anything wrong with my approach?

**Comment by amakropoulos** [comment, marked as answer]
Hi! Can you try setting the LLM Parallel Prompts parameter (Advanced Options) to the number of LLMCharacters, 2?
Probably the LLMCharacters are created after the LLM has started and it doesn't know that it has to handle more than 1.

**Reply by Moperius** [reply]
Nais! Great point. I went in and found this to be the problem.

One caveat - Since I'm instantiating prefabs of the LLM and Characters - If I change the parallelPrompts value to 2 after instantiation - it's still broken. Instantiating it as disabled, changing it to 2, then enabling - still broken. The value needs to be set before instantiation (in prefab) or it will break.

Thanks for the solution! 
o/
